rasa/core/nlg/response.py

```diff
@@ -24,29 +24,56 @@ def __init__(self, responses: Dict[Text, List[Dict[Text, Any]]]) -> None:
             responses: responses that will be used to generate messages.
         """
         self.responses = responses
+        print(self.responses)
+
+    def _matches_filled_slots(
+        self, filled_slots: Dict[Text, Any], response: Dict[Text, Any],
+    ) -> bool:
+        """Checks if the conditional response variation matches the filled slots."""
+        condition_list = response.get("condition")
+        for condition in condition_list:
+            name = condition["name"]
+            value = condition["value"]
+            if filled_slots.get(name) == value:
```

Comment:
this returns `True` as soon as one condition matches - what if there are multiple conditions?

Reply:
Very good point - changed the method a bit to support multiple constraints.